src/hints/execution.rs

```diff
@@ -1205,6 +1205,22 @@ pub fn set_ap_to_tx_nonce(
     Ok(())
 }
 
+pub const SET_FP_PLUS_4_TO_TX_NONCE: &str = "memory[fp + 4] = to_felt_or_relocatable(tx.nonce)";
+
+pub fn set_fp_plus_4_to_tx_nonce(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    _ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx: &InternalTransaction = exec_scopes.get_ref(vars::scopes::TX)?;
+    let nonce = tx.nonce.ok_or(HintError::AssertionFailed("tx.nonce should be set".to_string().into_boxed_str()))?;
+    vm.insert_value((vm.get_fp() + 4)?, nonce).map_err(HintError::Memory)?;
```

Comment:
Why do you `map_err`? `vm.insert_value(...)?` should be enough.

Reply:
I copied `insert_value_into_ap()` impl verbatim and replaced `ap` with `fp+4`. Seems unnecessary, though -- it's removed in a6f10f8